controller/service.rst

```diff
@@ -80,6 +80,118 @@ If your controller implements the ``__invoke()`` method - popular with the
 Action-Domain-Response (ADR) pattern, you can simply refer to the service id
 (``AppBundle\Controller\HelloController`` or ``app.hello_controller`` for example).
 
+If you want to use the ADR pattern rather than the default controller approach, you can use
+the new features provided by the container like public/private injections and autowiring, 
+in order to work, you must update the services.yml file ::
+
+    parameters:
+        #parameter_name: value
+
+    services:
+        _defaults:
+            autowire: true
+            autoconfigure: true
+            public: false
+            
+        # Allow to load every actions
+        AppBundle\Action\:
+            resource: '../../src/AppBundle/Action/'
+            public: true
+
+Once the file is updated, delete your Controller folder and create an Action one then 
+build a HelloAction class that use the ADR principes ::
+
+    <?php
+
+    namespace AppBundle\Action;
+
+    use Symfony\Component\HttpFoundation\Response;
+    use Symfony\Component\Templating\EngineInterface;
+
+    /**
+     * Class HelloAction
+     */
+    final class HelloAction
+    {
+        /** @var EngineInterface */
+        private $renderEngine;
+
+        /**
+         * HelloAction constructor.
+         *
+         * @param EngineInterface $renderEngine
+         */
+        public function __construct(
+            EngineInterface $renderEngine
+        ) {
+            $this->renderEngine = $renderEngine;
+        }
+
+        /**
+         * @throws \InvalidArgumentException
+         * @throws \RuntimeException
+         *
+         * @return Response
+         */
+        public function __invoke()
+        {
+            return new Response(
+                $this->renderEngine->render('default/index.html.twig')
+            );
+        }
+    }
+
+By default, we define the class with the final keyword because this class shouldn't been extended,
+the logic is pretty simple to understand as you understand the ADR pattern, in fact, the 'Action' 
+is linked to a single request and his dependencies are linked to this precise Action.
+
+Once this is done, you can define the routes like before using multiples approach :
+
+.. configuration-block::
+
+    .. code-block:: yaml
+    
+        # app/config/routing.yml
+        hello:
+            path:     /hello
+            methods: ['GET']
```

Comment:
This requirement has not been configured in the rest of formats so I guess can be removed?